src/components/ChooseLanguage.tsx

```diff
@@ -0,0 +1,101 @@
+import { createForm } from "@modular-forms/solid";
+import { useNavigate } from "@solidjs/router";
+import { createSignal, For, Show } from "solid-js";
+
+import { Button, ExternalLink, InfoBox, NiceP, VStack } from "~/components";
+import { useI18n } from "~/i18n/context";
+import { useMegaStore } from "~/state/megaStore";
+import {
+    Language,
+    eify,
+    timeout,
+    LANGUAGE_OPTIONS,
+    EN_OPTION
+} from "~/utils";
+
+type ChooseLanguageForm = {
+    selectedLanguage: string;
+};
+
+const COMBINED_OPTIONS: Language[] = [...LANGUAGE_OPTIONS];
+
+export function ChooseLanguage() {
+    const i18n = useI18n();
+    const [error, setError] = createSignal<Error>();
+    const [_state, actions] = useMegaStore();
+    const [loading, setLoading] = createSignal(false);
+    const navigate = useNavigate();
+
+    function findLanguageByValue(value: string) {
+        return (
+            COMBINED_OPTIONS.find((language) => language.value === value) ??
+            EN_OPTION
+        );
+    }
+
+    const [_chooseLanguageForm, { Form, Field }] =
+        createForm<ChooseLanguageForm>({
+            initialValues: {
+                selectedLanguage: ""
+            },
+            validate: (values) => {
+                const errors: Record<string, string> = {};
+                if (values.selectedLanguage === undefined) {
+                    errors.selectedLanguage = i18n.t(
+                        "settings.currency.error_unsupported_currency"
+                    );
+                }
+                return errors;
+            }
+        });
+
+    const handleFormSubmit = async (f: ChooseLanguageForm) => {
+        setLoading(true);
+        try {
+            actions.saveLanguage(findLanguageByValue(f.selectedLanguage).label);
+
+            await timeout(1000);
+            navigate("/");
```

Comment:
One issue I can't figure out is how to refresh the i18n object, since `navigate("/")` doesn't force a refresh I think a `window.location.reload()` could be an option unless there is some i18n api call I am missing to force a locale refresh

Reply:
Yep, I didn't find any specific i18n call, but I'll try with the windown.location.reload() then 🤔

Reply:
Ideally we don't want to have the page refresh, perhaps there is a way to make the i18nProvider component re-render on a language change

Reply:
Cool, I'll do a little bit more research about it then.

Reply:
@benthecarman gave me a little help and we figured out 🚀

Reply:
awesome works great!